tests/transforms/test_tape_expand.py

```diff
@@ -24,6 +24,15 @@
 from pennylane.wires import Wires
 
 
+@pytest.fixture(scope="function", autouse=True)
+def capture_warnings(recwarn):
+    yield
+    if len(recwarn) > 0:
+        for w in recwarn:
+            assert isinstance(w.message, qml.PennyLaneDeprecationWarning)
+            assert "'expansion_strategy' attribute is deprecated" in str(w.message)
```

Comment:
So the purpose of this fixture is to capture all `expansion_strategy` related deprecation warnings that are not caught by `pytest.warns` so that an error isn't raised when pytest sees `expansion_strategy` related deprecation warnings? I'm assuming this is so that you don't have to manually add `pytest.warns` to every single test that uses `expansion_strategy`? But why is this fixture specific to this file?

Reply:
Exactly! In principle, it doesn't need to live in this file, but I preferred to keeps things localized such that removal is easier by having everything collected in one spot. I am also adding a similar fixture for legacy devices so I kept this one here hoping that they won't conflict once one of the PRs is merged first.